hass_nabucasa/llm.py

```diff
@@ -0,0 +1,291 @@
+        tool_choice: Literal["auto", "none", "required"] | dict[str, Any] | None = None,
+    ) -> ResponsesAPIResponse | BaseResponsesAPIStreamingIterator:
+        """Generate structured or free-form LLM data."""
+        await self.async_ensure_token()
+
+        response_kwargs: dict[str, Any] = {
+            "model": self._generate_data_model,
+            "input": messages,
+            "api_key": self._token,
+            "api_base": self._base_url,
+            "user": conversation_id,
+            "stream": stream,
+            "response_format": response_format,
+            "tools": tools,
+            "tool_choice": tool_choice,
+        }
+
+        try:
+            response = await aresponses(**response_kwargs)
+            return cast(
+                "ResponsesAPIResponse | BaseResponsesAPIStreamingIterator", response
+            )
+        except AuthenticationError as err:
+            raise LLMAuthenticationError("Cloud LLM authentication failed") from err
+        except (RateLimitError, ServiceUnavailableError) as err:
+            raise LLMRateLimitError("Cloud LLM is rate limited") from err
+        except APIError as err:
+            raise LLMServiceError("Error talking to Cloud LLM") from err
+        except Exception as err:
+            raise LLMServiceError(
+                "Unexpected error during LLM data generation"
+            ) from err
+
+    async def async_generate_image(
+        self,
+        *,
+        prompt: str,
+        attachments: list[LLMImageAttachment] | None = None,
+    ) -> LLMGeneratedImage:
+        """Generate or edit an image via Cloud LLM."""
+        await self.async_ensure_token()
+
+        try:
+            if attachments:
+                return await self._async_edit_image(prompt, attachments)
+            return await self._async_create_image(prompt)
+        except AuthenticationError as err:
+            raise LLMAuthenticationError("Cloud LLM authentication failed") from err
+        except (RateLimitError, ServiceUnavailableError) as err:
+            raise LLMRateLimitError("Cloud LLM is rate limited") from err
+        except APIError as err:
+            raise LLMServiceError("Error talking to Cloud LLM") from err
+        except Exception as err:
+            raise LLMServiceError(
+                "Unexpected error during LLM data generation"
+            ) from err
+
+    async def _async_create_image(self, prompt: str) -> LLMGeneratedImage:
+        response = await aimage_generation(
+            prompt=prompt,
+            api_key=self._token,
+            api_base=self._base_url,
+            model=self._generate_image_model,
+        )
+
+        return await _extract_response_image_data(response)
+
+    async def _async_edit_image(
+        self, prompt: str, attachments: list[LLMImageAttachment]
+    ) -> LLMGeneratedImage:
+        if not self._generate_image_model:
+            raise LLMServiceError("Image editing model is not configured")
```

Comment:
If this path is valid, then the type of the response is wrong.
If this is for typing ant it does not know that we run ensure_token first use this instead:
```suggestion
        if TYPE_CHECKING:
            assert self._generate_image_model is not None
```